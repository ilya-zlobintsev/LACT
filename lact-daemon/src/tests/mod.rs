use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
