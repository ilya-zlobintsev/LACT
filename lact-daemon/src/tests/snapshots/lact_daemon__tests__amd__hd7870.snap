---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": null,
  "info": {
    "driver": "radeon",
    "link_info": {
      "current_speed": "8.0 GT/s PCIe",
      "current_width": "1"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Pitcairn XT [Radeon HD 7870 GHz Edition]",
        "model_id": "6818",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model": "Sapphire HD 7870",
        "model_id": "E217",
        "vendor": "PC Partner Limited / Sapphire Technology",
        "vendor_id": "174B"
      }
    },
    "vulkan_info": {
      "api_version": "",
      "device_name": "",
      "driver": {
        "version": 0
      },
      "enabled_layers": [],
      "extensions": {},
      "features": {}
    }
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Pitcairn XT [Radeon HD 7870 GHz Edition]",
      "model_id": "6818",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model": "Sapphire HD 7870",
      "model_id": "E217",
      "vendor": "PC Partner Limited / Sapphire Technology",
      "vendor_id": "174B"
    }
  },
  "power_profile_modes": null,
  "power_states": {
    "core": [],
    "vram": []
  },
  "stats": {
    "clockspeed": {
      "gpu_clockspeed": 300
    },
    "fan": {
      "control_enabled": false,
      "pmfw_info": {},
      "pwm_current": 51,
      "pwm_max": 255,
      "pwm_min": 0
    },
    "power": {},
    "temps": {
      "1": {
        "crit": 120.0,
        "crit_hyst": 90.0,
        "current": 27.0
      }
    },
    "voltage": {},
    "vram": {}
  }
}
